src/plugins/workspace/server/plugin.ts

```diff
@@ -64,12 +70,80 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
     });
   }
 
+  private async setupPermission(
+    core: CoreSetup,
+    { applicationConfig }: AppPluginSetupDependencies
+  ) {
+    this.permissionControl = new SavedObjectsPermissionControl(this.logger);
+
+    core.http.registerOnPostAuth(async (request, response, toolkit) => {
+      let groups: string[];
+      let users: string[];
+
+      // There may be calls to saved objects client before user get authenticated, need to add a try catch here as `getPrincipalsFromRequest` will throw error when user is not authenticated.
+      try {
+        ({ groups = [], users = [] } = this.permissionControl!.getPrincipalsFromRequest(request));
+      } catch (e) {
+        return toolkit.next();
+      }
+      if (groups.length === 0 && users.length === 0) {
+        updateWorkspaceState(request, {
+          isDashboardAdmin: false,
+        });
+        return toolkit.next();
+      }
+
+      this.logger.info('Dynamic application configuration enabled:' + !!applicationConfig);
+      if (!!applicationConfig) {
+        const [coreStart] = await core.getStartServices();
+        const scopeClient = coreStart.opensearch.client.asScoped(request);
+        const applicationConfigClient = applicationConfig.getConfigurationClient(scopeClient);
+
+        const [configGroups, configUsers] = await Promise.all([
+          applicationConfigClient
+            .getEntityConfig('opensearchDashboards.dashboardAdmin.groups')
+            .catch(() => undefined),
+          applicationConfigClient
+            .getEntityConfig('opensearchDashboards.dashboardAdmin.users')
+            .catch(() => undefined),
+        ]);
+
+        updateDashboardAdminStateForRequest(
+          request,
+          groups,
+          users,
+          configGroups ? [configGroups] : [],
+          configUsers ? [configUsers] : []
```

Comment:
```suggestion
          configGroups ? configGroups : [],
          configUsers ? configUsers : []
```
configGroups and configUsers should already be array I think?